cli/cmd/tap.go

```diff
@@ -292,6 +302,16 @@ func renderTapEvent(event *pb.TapEvent, resource string) string {
 	}
 }
 
+// renderTapEventJSON renders a Public API TapEvent to a string in JSON format.
+func renderTapEventJSON(event *pb.TapEvent, _ string) string {
+	e, err := json.MarshalIndent(event, "", "  ")
```

Comment:
Go's built-in JSON serializer doesn't work very well with Protobuf structs, so the Go implementation of protobuf actually provides its own serializer, here:

https://godoc.org/github.com/golang/protobuf/jsonpb#Marshaler

That marshaler is more configurable, and the JSON objects are flatter than those produced with the built-in serializer. Linkerd's web server is using it to send JSON payloads back to the web client, here:

https://github.com/linkerd/linkerd2/blob/8270ba363cc9c1750d63da2f745ad8c86d69c13f/web/srv/api_handlers.go#L36

And here:

https://github.com/linkerd/linkerd2/blob/8270ba363cc9c1750d63da2f745ad8c86d69c13f/web/srv/api_handlers.go#L62-L65